dhall/tests/Dhall/Test/TH.hs

```diff
@@ -1,14 +1,13 @@
 {-# LANGUAGE DeriveAnyClass     #-}
 {-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE DerivingStrategies #-}
```

Comment:
This makes Hydra fail with GHC 8.0.2:

```
tests/Dhall/Test/TH.hs:3:14: error:
    Unsupported extension: DerivingStrategies
```

https://hydra.dhall-lang.org/build/52074/nixlog/4

Reply:
Yeah, unfortunately, I can't use `CPP` to control the presence or absence of language extensions (I tried wrapping that in a `MIN_VERSION_…` guard and it failed)

There are three basic options I can think of:

* Disable the test suite for GHC 8.0.2
* Drop support for GHC 8.0.2
* Disable the use of `DerivingStrategies` in the generated code

Reply:
> Yeah, unfortunately, I can't use `CPP` to control the presence or absence of language extensions (I tried wrapping that in a `MIN_VERSION_…` guard and it failed)

What's the failure you're seeing? Having he language extension in CPP seems to work fine in the following snippet (tested with GHC 8.0.2 and 8.2.2):

```haskell
{-# language CPP #-}

#if MIN_VERSION_base(4,10,0)
{-# language DerivingStrategies #-}
#endif

module M where

data D = D
#if MIN_VERSION_base(4,10,0)
  deriving stock Show
#else
  deriving Show
#endif

main :: IO ()
main = print D
```

Reply:
@sjakobi: Never mind.  I figured out what I was doing wrong.